presto-main/src/main/java/com/facebook/presto/sql/gen/InputReferenceCompiler.java

```diff
@@ -106,4 +95,61 @@ public BytecodeNode visitVariableReference(VariableReferenceExpression reference
     {
         throw new UnsupportedOperationException();
     }
+
+    static class InputReferenceNode
+            implements BytecodeNode
+    {
+        IfStatement body;
+        BytecodeExpression block;
+        BytecodeExpression position;
+
+        public InputReferenceNode(CallSiteBinder callSiteBinder, Scope scope, Type type, BytecodeExpression block, BytecodeExpression position)
```

Comment:
Sorry I didn't realize the refactor I suggested also requires `callSiteBinder`, `scope` and `type` as input to the constructor. 

This constructor is not suited as a public constructor. Let's make it private (it can still be called by its outer class, `InputReferenceCompiler`